Sources/Tracing/NoOpTracer.swift

```diff
@@ -139,6 +139,12 @@ public struct NoOpTracer: LegacyTracer {
             }
         }
 
+        /// Update the span attributes in a block instead of individually.
+        ///
+        /// The NoOpSpan implementation does not call the closure as setting attributes
+        /// on a NoOpSpan does nothing.
+        public func updateAttributes(_ update: (inout SpanAttributes) -> Void) {}
```

Comment:
Hmm see my other comment, I still think we should call the closure, just with local empty span attributes that get discarded immediately, not saved on the instance.